torchprime/torch_xla_models/scan_layers.py

```diff
@@ -0,0 +1,55 @@
+import torch
+import torch.nn as nn
+from functorch.compile import default_partition
+from torch_xla.experimental.scan_layers import scan_layers
+
+from torchprime.layers.sequential import HomogeneousSequential, PyTree, splat
+
+
+class HomogeneousSequentialScan(HomogeneousSequential):
+  def __init__(self, *args, partition_fn=default_partition):
+    super().__init__(*args)
+    self.partition_fn = partition_fn
+
+  def forward(self, *input, **broadcasted_inputs: PyTree):
+    layers = [BroadcastArguments(m) for m in self.children()]
```

Comment:
Can you elaborate on what self.children() will refer to in this case? I can't figure it out by reading the code.

Reply:
Added a comment. It just returns the sequence of decoder layers. So this way we wrap each in a `BroadcastArguments`, which is a trick to make the decoder layers work with `scan`. In the `BroadcastArguments` we extend each decoder layer's output to also output the broadcasted inputs (position IDs, etc). This plumbs those values across scan iterations so the same position IDs are available to all layers.

Reply:
I see. It is coming from nn.Sequential that is the parent class of HomogeneousSequential 👍